src/ajax.js

```diff
@@ -462,14 +462,14 @@ jQuery.extend( {
 
 				// Raw string
 				getAllResponseHeaders: function() {
-					return state === 2 ? responseHeadersString : null;
+					return completed ? responseHeadersString : null;
 				},
 
 				// Caches the header
 				setRequestHeader: function( name, value ) {
-					var lname = name.toLowerCase();
-					if ( !state ) {
-						name = requestHeadersNames[ lname ] = requestHeadersNames[ lname ] || name;
+					if ( completed == null ) {
```

Comment:
Any thoughts on how to document it? I see what you mean about the style guide, and even though this _technically_ fits because I'd allow `null` to pass, the fact is that it will never come up so it's arguably against the spirit of our style guide. I could also just suck it up and use `=== undefined` if you think the current approach is confusing.